docs/dependency_confusion.md

```diff
@@ -1,140 +1,55 @@
-or other mechanism for pinning dependency versions. Cachi2 requires this. Verifying checksums is
-usually up to you.
+## Package Managers
 
-# Package Managers
+cachi2 follows some of the best practices, such as verifying the checksums of your dependencies
+to ensure their integrity or forcing you to pin all your dependencies to exact versions to avoid
+unexpected updates. However, it is essential to understand how a package manager works and
+what security features it provides to protect your software supply chain. cachi2 supports the
+following package managers:
 
-## gomod
+### gomod
 
-*TL;DR: commit your go.sum.*
+- package manager for Go
+- lock file: `go.sum`
+- [docs/gomod.md](https://github.com/containerbuildsystem/cachi2/tree/main/docs/gomod.md)
 
-Golang modules are defined by their [go.mod](https://golang.org/ref/mod#go-mod-file) files. These
-define the name of your module as well as the names and versions of its dependencies. Thanks to
-golang’s unique system of dependency management, this file is enough to ensure reproducible builds
-on its own.
+### pip
 
-However, due to golang versions being tied to git tags, you are technically at the mercy of
-repository maintainers. To prevent surprises, whenever go downloads any dependencies, it saves
-their checksums in a [go.sum](https://golang.org/ref/mod#go-sum-files) file. This file is later used
-to verify that the content of all direct and indirect dependencies is what you expect it to be.
+- package manager for Python
+- lock file: `requirements.txt`
+- [docs/pip.md](https://github.com/containerbuildsystem/cachi2/tree/main/docs/pip.md)
 
-As long as you make sure to always commit your go.sum file, your Golang module should be safe from
-dependency attacks.
+### npm
 
-## *[Cachito][cachito-1] also supports the following package managers, but Cachi2 does not (yet).*
+- package manager for JavaScript
+- lock file: `package-lock.json`
+- [docs/npm.md](https://github.com/containerbuildsystem/cachi2/tree/main/docs/npm.md)
 
-> ## pip
->
-> *TL;DR: if your package is Cachito compliant, it is most likely safe. If you want to be sure, verify
-> checksums.*
->
-> Python packages can have various formats, which are specified across multiple PEPs. Pip supports
-> most of them. Cachito support for pip, on the other hand, is quite simple. You must define all the
-> direct and indirect dependencies of your package in
-> [requirements.txt](https://pip.pypa.io/en/stable/user_guide/#requirements-files) style files and
-> tell Cachito to process them.
->
-> Cachito further restricts what you can put in your requirements.txt files. All of the dependencies
-> must be
-> [pinned](https://github.com/release-engineering/cachito/blob/master/docs/pip.md#pinning-versions)
-> to an exact version. Cachito will refuse to process requirements files that use the --index-url or
-> --extra-index-url options, which means private registries are out of the question. These two
```

Comment:
So for pip, apart from `index-url` everything else should be more or less true even now for cachi2.